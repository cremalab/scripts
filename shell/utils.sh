#!/bin/bash
function print {
  echo ""
  echo "-----------------------------------------------------"
  echo $1
  echo "-----------------------------------------------------"
}