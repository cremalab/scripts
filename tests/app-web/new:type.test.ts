import { cleanup } from "../utils/cleanup"
import { generateNewType } from "../utils/generateNewType"
import { navigateToConsumer } from "../utils/navigateToConsumer"
import { printTreeOfNew } from "../utils/printTreeOfNew"

describe("new:type", () => {
  beforeAll(navigateToConsumer("app-web"))
  afterAll(cleanup("./src/types/Foo.ts"))

  it("generates type file", async () => {
    await generateNewType()
    const tree = printTreeOfNew("types")
    expect(tree).toMatchInlineSnapshot(`
    "types
    ├── Foo.ts
    └── README.md
    "
    `)
  })
})
