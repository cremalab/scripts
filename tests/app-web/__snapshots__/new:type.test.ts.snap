// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new:type generates type file with expected contents 1`] = `
"/**
 * Foo
 * ------------------------------------------------
 * DESCRIPTION_HERE
 */

export interface Foo {}
"
`;
